components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -137,8 +132,9 @@ public boolean validateGrant(OAuthTokenReqMessageContext tokReqMsgCtx) throws Id
 
         String[] allowedScopes = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getScope();
         tokReqMsgCtx.setScope(allowedScopes);
-        tokReqMsgCtx.addProperty(OrganizationSwitchGrantConstants.TOKEN_BINDING_REFERENCE,
-                tokenDO.getTokenBinding().getBindingReference());
+        if (tokenDO.getTokenBinding() != null) {
```

Comment:
check why setting token binding from here not available when using issue method.

Reply:
Passing as a property is working fine. But the reason for token binding get replaced is not yet discovered.